File: apnaghar-frontend/src/services/paymentService.js
import api from "./api";

// ✅ Pay for booking
export const payForBooking = (bookingId) => {
  return api.put(`/payments/pay/${bookingId}`);
};

// ✅ Get my payments
export const getMyPayments = () => {
  return api.get("/payments/my");
};

File: src/main/java/com/apnaghar/service/PaymentService.java
package com.apnaghar.service;

import java.util.List;
import com.apnaghar.model.Payment;

public interface PaymentService {

    // ✅ USER PAYS FOR BOOKING
    Payment payForBooking(Long bookingId, String userEmail);

    // ✅ USER PAYMENT HISTORY
    List<Payment> getMyPayments(String userEmail);
    
    //get owner earning 
    List<Payment> getOwnerPayments(String ownerEmail);

}
